src/opers.c

```diff
@@ -1981,2769 +1952,627 @@ static inline Obj CacheOper (
-          }
-        }
-      
-      /* otherwise try to find one in the list of methods                    */
-      if (!method)
-        {
-          if (prec == INTOBJ_INT(0))
-            method = CALL_1ARGS( Method0Args, oper );
-          else
-            method = CALL_2ARGS( NextMethod0Args, oper, prec );
-
-          /* If there was no method found, then pass the information needed for
-             the error reporting. This function rarely returns */
-          
-          while (method == Fail)
-            method = CallHandleMethodNotFound( oper, 0, (Obj *) 0, 0, 0, prec);
-          
-          /* update the cache */
-          if (method && prec < INTOBJ_INT(CACHE_SIZE))
-            {
-              Bag cacheBag = GET_METHOD_CACHE( oper, 0 );
-              cache = 1+ADDR_OBJ( cacheBag );
-              cache[2*STATE(CacheIndex)] = method;
-              cache[2*STATE(CacheIndex)+1] = prec;
-              STATE(CacheIndex) = (STATE(CacheIndex) + 1) % CACHE_SIZE;
-              CHANGED_BAG( cacheBag );
-            }
-#ifdef COUNT_OPERS
-          OperationMiss++;
-#endif
-        }
-      if ( !method )  {
-        ErrorQuit( "no method returned", 0L, 0L );
-      }
-      
-      /* call this method                                                    */
-      res = CALL_0ARGS( method );
-    }
-    while (res == TRY_NEXT_METHOD );
-
-    /* return the result                                                   */
-    return res;
-}
-
-
-/****************************************************************************
-**
-**  DoOperation1Args( <oper>, <a1> )
-*/
-Obj DoOperation1Args (
-    Obj                 oper,
-    Obj                 arg1 )
-{
-    Obj                 res;
-    Obj                 type1;
-    Obj                 id1;
-    Obj *               cache;
-    Obj                 method;
-    Int                 i;
-    Obj                 prec;
-
-    /* get the types of the arguments                                      */
-    type1 = TYPE_OBJ_FEO( arg1 );
-    id1 = ID_TYPE( type1 );
-
-    /* try to find an applicable method in the cache                       */
-    prec = INTOBJ_INT(-1);
 
-    do {
-      /* The next line depends on the implementation of INTOBJS */
-      prec = (Obj)(((Int)prec) +4);
-      method = 0;
-
-      /* recalculate cache each pass, in case of GC     */
-      cache = 1+ADDR_OBJ( CacheOper( oper, 1 ) );
-
-      /* Up to CACHE_SIZE methods might be in the cache */
-      if (prec < INTOBJ_INT(CACHE_SIZE))
-        {
-          for (i = 0;  i < 3*CACHE_SIZE; i+= 3) {
-            if (  cache[i+1] == prec && cache[i+2] == id1 ) {
-              method = cache[i];
+static ALWAYS_INLINE Obj GetMethodCached(Obj  oper,
+                                         UInt n,
+                                         Int  prec,
+                                         Obj  ids[])
+{
+    UInt  typematch;
+    Obj * cache;
+    Obj   method = 0;
+    UInt  i;
+    UInt  CacheEntrySize = n + 2;
```

Comment:
`CacheEntrySize` is a constant, isn't? So perhaps add `const` (the compiler probably won't care, but it helps a human trying to understand the code. I.e. ;-).

Also, since it's a local variable, I'd change it to `cacheEntrySize`.